core/ns.go

```diff
@@ -59,14 +59,17 @@ func (ns *Namespace) AlterMeta(fn *Fn, args []Object) Map {
 }
 
 func (ns *Namespace) Hash() uint32 {
-	return HashPtr(uintptr(unsafe.Pointer(ns)))
+	return ns.hash
 }
 
+const nsHashMask uint32 = 0x90569f6f
+
 func NewNamespace(sym Symbol) *Namespace {
 	return &Namespace{
 		Name:     sym,
 		mappings: make(map[*string]*Var),
 		aliases:  make(map[*string]*Namespace),
+		hash:     sym.Hash() ^ nsHashMask,
```

Comment:
Is `^ nsHashMask` part there just to make the namespace's hash different from symbol's hash?

Reply:
Yes. There's an existing such mask being added for `(Symbol)Hash()`. (Want me to convert that into a similar form as I'm using above, e.g. make a var for it and use `^` instead of `+`, for consistency?)